oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/OIDCClientConfiguration.java

```diff
@@ -799,31 +804,44 @@ public boolean isSkipped()
      */
     public OIDCClaimsRequest getClaimsRequest()
     {
-        // TODO: allow passing the complete JSON as configuration
-        OIDCClaimsRequest claimsRequest = new OIDCClaimsRequest();
-
-        // ID Token claims
-        List<String> idtokenclaims = getIDTokenClaims();
-        if (idtokenclaims != null && !idtokenclaims.isEmpty()) {
-            ClaimsSetRequest idtokenclaimsRequest = new ClaimsSetRequest();
-
-            for (String claim : idtokenclaims) {
-                idtokenclaimsRequest = idtokenclaimsRequest.add(claim);
+        // Parse the complete claims JSON if configured
+        String claimsJson = getProperty(PROP_CLAIMS, String.class);
+        OIDCClaimsRequest claimsRequest = null;
+        if (claimsJson != null && claimsJson.trim().length() > 0) {
+            try {
+                claimsRequest = OIDCClaimsRequest.parse(claimsJson);
+            } catch (ParseException e) {
+                this.logger.warn("Parsing claims JSON " + claimsJson + "failed with message:" + e.getMessage());
```

Comment:
Actually with slf4j you are supposed to use a parameter replacement syntax (and the `[]` around values is more of an XWiki convention), also I would pass the root cause message instead of the direct exception (it's generally more useful) so something like:

```
this.logger.warn("Parsing claims JSON [{}] failed with message: {}", claimsJson, ExceptionUtils.getRootCauseMessage(e));
```

Reply:
Fixed